x-pack/plugins/upgrade_assistant/public/application/components/overview/identify_step/use_deprecation_logging.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { useAppContext } from '../../../app_context';
+import { ResponseError } from '../../../lib/api';
+import { DeprecationLoggingPreviewProps } from '../../types';
+
+const i18nTexts = {
+  enabledMessage: i18n.translate(
+    'xpack.upgradeAssistant.overview.deprecationLogs.enabledToastMessage',
+    {
+      defaultMessage: 'Log deprecated actions.',
+    }
+  ),
+  disabledMessage: i18n.translate(
+    'xpack.upgradeAssistant.overview.deprecationLogs.disabledToastMessage',
+    {
+      defaultMessage: 'Do not log deprecated actions.',
+    }
+  ),
+};
+
+export const useDeprecationLogging = (): DeprecationLoggingPreviewProps => {
+  const { api, notifications } = useAppContext();
+  const { data, error: fetchError, isLoading, resendRequest } = api.useLoadDeprecationLogging();
+
+  const [isEnabled, setIsEnabled] = useState(false);
```

Comment:
If we implement the suggestions I had for server-side naming, I think we should allow that change to ripple to the client too. This would result in this changing to:

```js
const [isDeprecationLoggingEnabled, setIsDeprecationLoggingEnabled] = useState(false);
```